File: 3536. Maximum Product of Two Digits.cpp
class Solution {
public:
    int maxProduct(int n) 
    {
        auto s = to_string(n);
        sort(begin(s), end(s));
        return ((s[s.size() - 1] - '0') * (s[s.size() - 2] - '0')); 
    }
};

File: 3536. Maximum Product of Two Digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        s = str(n)
        max_ch = max(s)
        if s.count(max_ch) > 1:
            return int(max_ch) * int(max_ch)

        s = s.replace(max_ch, '0')
        return int(max(s)) * int(max_ch)